File: 07-Healthcare-DevSecOps-Governance/services/data-ingestion/app/models/sources.py
from enum import StrEnum

from pydantic import BaseModel, Field


class AccessTier(StrEnum):
    synthetic = "SYNTHETIC"
    public_deidentified = "PUBLIC_DEIDENTIFIED"
    controlled_research = "CONTROLLED_RESEARCH"
    live_phi = "LIVE_PHI"


class EthicalSource(BaseModel):
    source_id: str
    name: str
    modalities: list[str]
    access_tier: AccessTier
    license: str
    homepage: str
    attribution_required: bool = False
    reidentification_prohibited: bool = True
    credential_required: bool = False
    dua_required: bool = False
    enabled: bool = True
    approval_reference: str | None = None
    restrictions: list[str] = Field(default_factory=list)


class SourceAssessment(BaseModel):
    allowed: bool
    source: EthicalSource
    reasons: list[str] = Field(default_factory=list)

File: 07-Healthcare-DevSecOps-Governance/portal/lib/api/sources.ts
import { api, unwrap } from "./client";

export interface EthicalSource {
  source_id: string;
  name: string;
  modalities: string[];
  access_tier: "SYNTHETIC" | "PUBLIC_DEIDENTIFIED" | "CONTROLLED_RESEARCH" | "LIVE_PHI";
  license: string;
  enabled: boolean;
  credential_required: boolean;
  dua_required: boolean;
  restrictions: string[];
}

export const sourceApi = {
  list: () => unwrap<{ items: EthicalSource[] }>(api.get("/ingestion/sources")),
};
